src/main/java/frc/robot/end_effector/EndEffectorIOVortex.java

```diff
@@ -0,0 +1,106 @@
+import com.revrobotics.spark.config.SparkFlexConfig;
+
+import au.grapplerobotics.LaserCan;
+import au.grapplerobotics.interfaces.LaserCanInterface;
+import edu.wpi.first.wpilibj.DigitalInput;
+
+public class EndEffectorIOVortex implements EndEffectorIO {
+        private final SparkFlex coralIntake = new SparkFlex(EndEffectorConstants.coralIntakeCAN, MotorType.kBrushless);
+        private final RelativeEncoder coralIntakeEncoder = coralIntake.getEncoder();
+        private final SparkFlex algaeIntake = new SparkFlex(EndEffectorConstants.algaeIntakeCAN, MotorType.kBrushless);
+        private final RelativeEncoder algaeIntakeEncoder = algaeIntake.getEncoder();
+        private final SparkFlex algaeWrist = new SparkFlex(EndEffectorConstants.algaeWristCAN, MotorType.kBrushless);
+        private final RelativeEncoder algaeWristEncoder = algaeWrist.getEncoder();
+        private final DigitalInput coralIntakeBeamBreakDigitalInput = new DigitalInput(EndEffectorConstants.coralIntakeBeamBreakDIO); 
+        private final static LaserCan LaserCanSensor = new LaserCan(EndEffectorConstants.AlgaeIntakeLaserCAN); 
+        public static double laserDistance = -1.0; //Default measurement indicating no measurement yet
+    public EndEffectorIOVortex () {
+
+        SparkFlexConfig coralIntakeconfig = new SparkFlexConfig();
+        SparkFlexConfig algaeIntakeconfig = new SparkFlexConfig();
+        SparkFlexConfig algaeWristconfig = new SparkFlexConfig();
+
+
+        coralIntakeconfig.idleMode(IdleMode.kBrake).smartCurrentLimit(EndEffectorConstants.coralIntakecurrentLimit).voltageCompensation(12.0);
+        algaeIntakeconfig.idleMode(IdleMode.kBrake).smartCurrentLimit(EndEffectorConstants.algaeIntakecurrentLimit).voltageCompensation(12.0);
+        algaeWristconfig.idleMode(IdleMode.kBrake).smartCurrentLimit(EndEffectorConstants.algaeWristcurrentLimit).voltageCompensation(12.0);
+
+        coralIntakeconfig.encoder
+            .positionConversionFactor(2.0 * Math.PI)
+            .velocityConversionFactor((2.0 * Math.PI) / 60.0)
+            .uvwMeasurementPeriod(10)
+            .uvwAverageDepth(2);
+        algaeIntakeconfig.encoder
+            .positionConversionFactor(2.0 * Math.PI)
+            .velocityConversionFactor((2.0 * Math.PI) / 60.0)
+            .uvwMeasurementPeriod(10)
+            .uvwAverageDepth(2);
+        algaeWristconfig.encoder
+            .positionConversionFactor(2.0 * Math.PI)
+            .velocityConversionFactor((2.0 * Math.PI) / 60.0)
+            .uvwMeasurementPeriod(10)
+            .uvwAverageDepth(2);
+
+        coralIntake.configure(coralIntakeconfig, ResetMode.kResetSafeParameters, PersistMode.kPersistParameters);
+        algaeIntake.configure(algaeIntakeconfig, ResetMode.kResetSafeParameters, PersistMode.kPersistParameters);
+        algaeWrist.configure(algaeWristconfig, ResetMode.kResetSafeParameters, PersistMode.kPersistParameters);
+        LaserCan.Measurement measurement = LaserCanSensor.getMeasurement();
```

Comment:
This should be happening in `updateIputs()`. You need to configure the LaserCAN like it's done here:

```java
  @Override
  public void robotInit() {
    lc = new LaserCan(0);
    // Optionally initialise the settings of the LaserCAN, if you haven't already done so in GrappleHook
    try {
      lc.setRangingMode(LaserCan.RangingMode.SHORT);
      lc.setRegionOfInterest(new LaserCan.RegionOfInterest(8, 8, 16, 16));
      lc.setTimingBudget(LaserCan.TimingBudget.TIMING_BUDGET_33MS);
    } catch (ConfigurationFailedException e) {
      System.out.println("Configuration failed! " + e);
    }
  }
  ```